peer/abstractlist/list.go

```diff
@@ -548,31 +557,38 @@ func (pl *List) newNoContextDeadlineError() error {
 }
 
 func (pl *List) newUnavailableError(err error) error {
-	return yarpcerrors.Newf(yarpcerrors.CodeUnavailable, "%q peer list timed out waiting for peer: %s", pl.name, err.Error())
+	return yarpcerrors.Newf(yarpcerrors.CodeUnavailable, "%q peer list timed out waiting for peer, %s assigned, fail-fast is not enabled: %s", pl.name, pl.formatNumPeers(), err.Error())
```

Comment:
little nits:
 - The formatting of this message is inconsistent with the "fail-fast is enabled" message. This message has # assigned after fail-fast status, while the other message has fail-fast status before # assigned
 - While I can guess what "assigned" peers means in this context, I'm not sure it's user-friendly
 - I still think separating out the 0 peers case is more useful than mixing the messages and just having the number be 0 or 1

Looking at the stakeholder experience, I think the scenarios/error groupings are
 - No peers in the peer list. We waited for a peer to show up but timed out (fail-fast is not enabled)
 - No peers in the peer list, and we didn't wait (fail-fast enabled)
 - There are X peers in the peer-list, but none of them are usable, and we timed out waiting. We need to also find a good word for "usable" and help the user understand why they may not be usable
 - Same as above, but we didn't wait (fail-fast enabled).

Reply:
We could just say “connected” or possibly “responsive”.